python-template/{{cookiecutter.placeholder_repo_name}}/Makefile

```
@@ -0,0 +1,67 @@
+.PHONY: clean compile_translations coverage diff_cover docs dummy_translations \
+        extract_translations fake_translations help \
+        quality requirements selfcheck test test-all upgrade validate
+
+.DEFAULT_GOAL := help
+
+# For opening files in a browser. Use like: $(BROWSER)relative/path/to/file.html
+BROWSER := python -m webbrowser file://$(CURDIR)/
+
+help: ## display this help message
+	@echo "Please use \`make <target>' where <target> is one of"
+	@awk -F ':.*?## ' '/^[a-zA-Z]/ && NF==2 {printf "\033[36m  %-25s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST) | sort
+
+clean: ## remove generated byte code, coverage reports, and build artifacts
+	find . -name '__pycache__' -exec rm -rf {} +
+	find . -name '*.pyc' -exec rm -f {} +
+	find . -name '*.pyo' -exec rm -f {} +
+	find . -name '*~' -exec rm -f {} +
+	coverage erase
+	rm -fr build/
+	rm -fr dist/
+	rm -fr *.egg-info
+
+coverage: clean ## generate and view HTML coverage report
+	pytest --cov-report html
+	$(BROWSER)htmlcov/index.html
+
+docs: ## generate Sphinx HTML documentation, including API docs
+	tox -e docs
+	$(BROWSER)docs/_build/html/index.html
+
+# Define PIP_COMPILE_OPTS=-v to get more information during make upgrade.
+PIP_COMPILE = pip-compile --rebuild --upgrade $(PIP_COMPILE_OPTS)
+
+upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
+upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+	pip install -qr requirements/pip-tools.txt
+	# Make sure to compile files after any other files they include!
+	$(PIP_COMPILE) -o requirements/pip-tools.txt requirements/pip-tools.in
```

```suggestion
	$(PIP_COMPILE) -o requirements/pip-tools.txt requirements/pip-tools.in
	pip install -r requirements/pip-tools.txt
```
I actually recommend installing pip *again* so that the version of pip-tools you pin is the version that will be used to upgrade things.

---

:+1: to this—we've had a couple cases of where a `make upgrade` results in upgrading pip, but a *successive* call to `make upgrade` would fail, since the new version breaks but the old version of pip is still in use.